### PR TITLE
Maven dependency checks

### PR DESCRIPTION
**Pull request by yschimke**
Towards pandering to maven's horrible version conflict strategy

https://jakewharton.com/nonsensical-maven-is-still-a-gradle-problem/

### CODE REVIEW

**Comment by yschimke** [comment]
Not failing uniformly but

```
> Task :okcurl:processResources
[compileClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.22 version changed: 1.9.22 → 1.9.23

[runtimeClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.22 version changed: 1.9.22 → 1.9.23
> Task :okcurl:sympathyForMrMaven FAILED
```

**Comment by yschimke** [comment]
```
[apiDependenciesMetadata] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24

[compileClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
> Task :okhttp-idna-mapping-table:sympathyForMrMaven FAILED
[implementationDependenciesMetadata] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[runtimeClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[testCompileClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[testImplementationDependenciesMetadata] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[testRuntimeClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
```

File: build.gradle.kts [hunk 1]
```diff
@@ -88,6 +89,7 @@ subprojects {
   apply(plugin = "checkstyle")
   apply(plugin = "ru.vyarus.animalsniffer")
   apply(plugin = "biz.aQute.bnd.builder")
+  apply(plugin = "io.github.usefulness.maven-sympathy")
```
Comment (JakeWharton) [inline build.gradle.kts:92]: May be worth only applying to modules which are published, but if it's not too invasive as-is don't worry about it.

**Review by JakeWharton — APPROVED** [review]

**Comment by yschimke** [comment]
It feels like it may be impossible to guarantee we can always resolve and make a "clean" release.

```
> Task :okhttp-idna-mapping-table:sympathyForMrMaven FAILED
[apiDependenciesMetadata] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[compileClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[implementationDependenciesMetadata] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[runtimeClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[testCompileClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[testImplementationDependenciesMetadata] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
[testRuntimeClasspath] dependency org.jetbrains.kotlin:kotlin-stdlib:1.9.23 version changed: 1.9.23 → 1.9.24
```

KotlinPoet 1.7.0 bumped to 1.9.24, but Okio is on 1.9.21

https://github.com/square/kotlinpoet/blob/1.17.0/gradle/libs.versions.toml

https://github.com/square/okio/blob/parent-3.9.0/build-support/build.gradle.kts

**Comment by yschimke** [comment]
@JakeWharton so ahead of Okio is ok? Just not behind?

**Comment by JakeWharton** [comment]
Yep. We always want to be greater than or equal to. Never less than.

**Pull request merged**